recipes/qpid-proton/all/conanfile.py

```diff
@@ -0,0 +1,154 @@
+from conan import ConanFile
+from conan.tools.build import check_min_cppstd
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+from conan.tools.files import copy, get, rmdir, export_conandata_patches, apply_conandata_patches
+from conan.tools.microsoft import is_msvc
+
+import os
+
+required_conan_version = ">=2.1.0"
+
+
+class QpidProtonConan(ConanFile):
+    name = "qpid-proton"
+    description = "Qpid Proton is a high-performance, lightweight messaging library."
+    license = "Apache-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://qpid.apache.org/proton/"
+    topics = ("conan", "qpid", "proton", "messaging", "message", "queue", "topic", "mq", "amqp")
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True
+    }
+
+    @property
+    def _min_cppstd(self):
+        return "14"
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        if not is_msvc(self):
+            self.requires("openssl/[>=1.1 <4]")
+        if self.settings.os == "Macos":
+            self.requires("libuv/1.49.2")
```

Comment:
Maybe a version range?
```suggestion
            self.requires("libuv/[>=1 <2]")
```